src-tests/test-rpc.lisp

```diff
@@ -7,87 +7,110 @@
 
 (defparameter *expected-response* "test-response")
 
-(defun test-method (&key (sleep 1 sleep-p))
-  (when sleep-p
+(defun test-method (&key (sleep 1 slee-p))
+  (when slee-p
     (sleep sleep))
   "test-response")
 
+(defmacro with-unique-rpc-address ((addr) &body body)
```

Comment:
At a cursory glance, it seems like maybe this macro could be rolled up into (or maybe complementary to) your recent `with-random-rpc-client` over in rigetti/quilc#288.  I think at minimum `with-random-rpc-client` would need to be extended to take an (optional) `server-function` argument to support both the use case over in rigetti/quilc#288 and the use case below (but I haven't looked below too closely yet).

If so, that would be another case in favor of some kind of common "utils" package as advocated in rigetti/quilc#246

Just mentioning it here so it shows up over in rigetti/quilc#246, in case that ever comes to fruition.

Reply:
I actually think this could belong as a utility in rpcq itself.

Reply:
Sorry: "this" for me meant quilc's `with-random-rpc-client`. That's what I think could belong here.

Reply:
^ Good point

Reply:
FWIW I had to clone CCL [from github](https://github.com/Clozure/ccl) and use their [latest binaries](https://github.com/Clozure/ccl/releases/tag/v1.12-dev.5). (Their build system is weird, so in case it's not clear for you [it wasn't for me], you have to put those binaries in the same location as the git repo.)

Reply:
Installing the latest dev snapshot did the trick. Thanks for the tip.

Reply:
WRT general macro: I've raised the `with-random-rpc-client` to be exported from the `rpcq` package. I don't think we should make this more general and try to allow for the functionality expressed in the quilc rpc tests -- quilc should maintain it's own specific macro, on top of `with-random-rpc-client`.

Reply:
Part of the reason I mentioned extending `with-random-rpc-client` is that I personally feel like `with-unique-rpc-address` doesn't quite do enough to justify needing a macro, since it's a pretty thin wrapper around `let` (which is not the case for `with-random-rpc-client`).

In the case of `with-unique-rpc-address`, you could get almost as much joy out of a function like

```lisp
(defun make-inproc-url (&optional (address (uuid:make-v4-uuid)))
  (format nil "inproc://~a" address))
```

then let the caller handle the let-binding themselves.

This is just a minor personal preference though, not a blocker for this PR.

Reply:
I humbly disagree. It's thin but it's useful. Some utility of the `with-` macros is to remove the overhead of managing bindings. Like you said, with `make-inproc-url` you'd have to bind stuff. I agree that there is an opportunity for the quilc macro to be improved, but I don't think it should belond in RPCQ.

Reply:
Fair enough

Reply:
Actually, I've had lunch and now I've changed my mind. But I would like to GTM, so it's something for another day.